src/distribution_spatial.cpp

```diff
@@ -180,6 +184,102 @@ Position SphericalIndependent::sample(uint64_t* seed) const
   return {x, y, z};
 }
 
+//==============================================================================
+// MeshIndependent implementation
+//==============================================================================
+
+// Loosely adapted Patrick's code shown here
+MeshIndependent::MeshIndependent(pugi::xml_node node)
+{
+  // No in-tet distributions implemented, could include distributions for the barycentric coords
+  // Read in unstructured mesh from mesh_id value
+  int32_t mesh_id = std::stoi(get_node_value(node, "mesh_id"));
+  mesh_map_idx_ = model::mesh_map[mesh_id];
+  mesh_map_idx_ = model::mesh_map.at(mesh_id);
+  const auto& mesh_ptr = model::meshes[mesh_map_idx_];
+  
+  // Check whether mesh pointer points to an unstructured mesh
+  umesh_ptr_ = dynamic_cast<UnstructuredMesh*>(mesh_ptr.get());
+  if (!umesh_ptr_) {fatal_error("Mesh passed to spatial distribution is not an unstructured mesh object"); }
+
+  // Create CDF based on weighting scheme
+  int64_t tot_bins = umesh_ptr_->n_bins();
+  write_message(std::to_string(tot_bins));
+  float weights[tot_bins] = {};
+  float temp_CDF[tot_bins+1] = {0.0};
+  float temp_total_weight = 0.0;
+  mesh_CDF_.resize(tot_bins+1);
+  mesh_CDF_[0] = {0.0};
+  total_weight_ = 0.0;
+  int i = 0;
+
+  // Create cdfs for sampling for an element over a mesh
+  // Equal scheme is an unweighted sampling over the mesh
+  // Volume scheme is weighted by the volume of each tet
+  // File scheme is weighted by an array given in the xml file
+
+  if (check_for_node(node, "elem_weight_scheme")) {
+    sample_scheme_ = get_node_value(node, "elem_weight_scheme");
+  } else {
+      fatal_error("No weighting scheme was specified for element sampling over the mesh");
+  }
+
+  if (sample_scheme_ == "equal"){
```

Comment:
I think we had talked about removing the option for an equal weighting scheme. Where do you stand on that?

Reply:
If there is no use case for equal weighting I can remove it. I thought it may still be useful down the line (maybe with some sort of VR weighting) to get better statistics in a fine area of the mesh or just for testing. These may not be important/maybe it's niche enough where you would just use file sources.